ciao-controller/openstack_volume.go

```diff
@@ -297,14 +297,17 @@ func (c *controller) ListVolumesDetail(tenant string) ([]block.VolumeDetail, err
 		vol.ID = data.ID
 		vol.Size = data.Size
 		vol.OSVolTenantAttr = data.TenantID
-		vol.CreatedAt = &data.CreateTime
+		createdAt := data.CreateTime
+		vol.CreatedAt = &createdAt
```

Comment:
I get slightly confused after so many years in a non-garbage collected language, but:

This means we make a copy, then store a link to that copy.  Repeatedly.  But I assume the structure has pointers to avoid copies.  Would it then be more direct to not have the block.VolumeDetail structure use pointers for these members?